cmd/overlay.go

```diff
@@ -0,0 +1,147 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/speakeasy-api/openapi-specedit/pkg/overlay"
+	"gopkg.in/yaml.v3"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/speakeasy-api/openapi-specedit/pkg/loader"
+)
+
+var overlayCmd = &cobra.Command{
+	Use:   "overlay",
+	Short: "Work with OpenAPI Overlays",
+}
+
+var overlayValidateCmd = &cobra.Command{
+	Use:   "validate",
+	Short: "Given an overlay, it will state whether it appears to be valid according to the OpenAPI Overlay specification",
+	Args:  cobra.NoArgs,
+	RunE:  RunValidateOverlay,
+}
+
+var overlayCompareCmd = &cobra.Command{
+	Use:   "compare",
+	Short: "Given two specs, it will output an overlay that describes the differences between them",
+	Args:  cobra.NoArgs,
+	RunE:  RunCompare,
+}
+
+var overlayApplyCmd = &cobra.Command{
+	Use:   "apply",
+	Short: "Given an overlay, it will apply it to the spec. If omitted, spec will be loaded via extends (only from local file system).",
+	Args:  cobra.NoArgs,
+	RunE:  RunApply,
+}
+
+func overlayInit() {
+	overlayCmd.AddCommand(overlayValidateCmd)
+	overlayCmd.AddCommand(overlayCompareCmd)
+	overlayCmd.AddCommand(overlayApplyCmd)
+
+	overlayValidateCmd.Flags().StringP("overlay", "o", "", "overlay file to validate")
+	overlayValidateCmd.MarkFlagRequired("overlay")
+
+	overlayCompareCmd.Flags().StringSliceP("schemas", "s", []string{}, "schemas to compare and generate overlay from")
```

Comment:
In my experience the input for string slice in cobra is not very user friendly -s file1 -s file2

Might be better just to do a comma separate string of files and separate them here. That's what we've done in some other places

Reply:
I'm not sure I understand your comment. `StringSlice` generates `[]string{"a", "b", "c"}` in all of the following cases:

```sh
cmd -s a -s b -s c
cmd -s a,b -s c
cmd -s a,b,c
cmd -s="a,b" -s="c"
```

I prefer it over `StringArray` because it enables all those, whereas `StringArray` only works with the first option.

Reply:
I would prefer to use positional args here, but I would like to fix the interactive UI to make it handle positionals better than it does current if I'm going to do that.